var/spack/repos/builtin/packages/bwa/package.py

```diff
@@ -38,6 +38,10 @@ def install(self, spec, prefix):
                     'Makefile')
         # use spack C compiler
         filter_file('^CC=.*', 'CC={0}'.format(spack_cc), 'Makefile')
+        # fix gcc 10+ errors
+        if self.spec.satisfies('%gcc@10:'):
+            filter_file(r'const uint8_t rle_auxtab\[8\]',
+                        'extern const uint8_t rle_auxtab[8]', 'rle.h')
```

Comment:
To get rid of the escaping, you can pass `string=True` so the first parameter is not interpreted as a regular expression.